util/typeutil/stringInt_test.go

```diff
@@ -0,0 +1,28 @@
+package typeutil
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnmarshalJSON(t *testing.T) {
+	type Item struct {
+		ItemId StringInt `json:"item_id"`
+	}
+
+	t.Run("happy path - convert string to int", func(t *testing.T) {
+		jsonData := []byte(`{"item_id":"30"}`)
+		var item Item
+		assert.NoError(t, json.Unmarshal(jsonData, &item))
+		assert.Equal(t, 30, int(item.ItemId))
+	})
+
+	t.Run("happy path - make sure integer still works", func(t *testing.T) {
+		jsonData := []byte(`{"item_id":30}`)
+		var item Item
+		assert.NoError(t, json.Unmarshal(jsonData, &item))
+		assert.Equal(t, 30, int(item.ItemId))
+	})
+}
```

Comment:
Please include error conditions, such as a string which cannot be converted to an integer or a different incompatible type, like a boolean.

Reply:
this is done - https://github.com/prebid/prebid-server/pull/2617/commits/9e4089964012932b327478fc68b37b411a51bdcf